test/batch_test.cc

```diff
@@ -0,0 +1,38 @@
+#include "benchmark/benchmark.h"
+
+#include <algorithm>
+#include <cassert>
+
+static void BM_batch_iteration_count(benchmark::State& state) {
+  const size_t batch_size = state.range(0);
+  size_t actual_iterations = 0;
+  while (state.KeepRunningBatch(batch_size)) {
+    actual_iterations += batch_size;
+  }
+  assert(state.iterations() >= state.max_iterations);
```

Comment:
Was following along with all the other tests which use asserts. Tried it and we don't have visibility of check.h in test/